scripts/validate-oyd-vnet.sh

```diff
@@ -0,0 +1,394 @@
+#!/bin/bash -e
+
+function get_resource() {
+    resource_id=$1
+    api_version=$2
+
+    access_token=$(az account get-access-token --resource https://management.azure.com --query "accessToken" --output tsv)
+
+    curl -s -X GET https://management.azure.com$resource_id?api-version=$api_version -H "Authorization: Bearer $access_token" -H "Accept: application/json"
```

Comment:
https://learn.microsoft.com/en-us/cli/azure/reference-index?view=azure-cli-latest#az-rest

az rest is preferred comparing to curl, in case the user doesn't have curl installed, also make the line shorter by drop the header token and ARM host

Reply:
Updated!